python/ray/data/_internal/planner/map_batches.py

```diff
@@ -51,17 +51,29 @@ def validate_batch(batch: Block) -> None:
                 )
 
             if isinstance(batch, collections.abc.Mapping):
-                for key, value in batch.items():
-                    if not isinstance(value, np.ndarray):
+                for key, value in list(batch.items()):
```

Comment:
I'm curious why you need the `list(` here -- shouldn't `.items()` always return an object that can be iterated on with the `for key, value in batch.items()` pattern?

Reply:
This was to avoid mutating the dict we are iterating over. Though, maybe in this case it is ok, since it doesn't change the keys.